spark-connector/spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/GravitinoIcebergCatalog.java

```diff
@@ -157,6 +159,22 @@ public org.apache.spark.sql.connector.catalog.Table loadTable(Identifier ident,
     }
   }
 
+  @Override
+  protected String getDatabase(Identifier sparkIdentifier) {
+    if (sparkIdentifier.namespace() == null || sparkIdentifier.namespace().length == 0) {
+      return getCatalogDefaultNamespace();
+    } else if (sparkIdentifier.namespace().length == 1) {
+      return sparkIdentifier.namespace()[0];
+    } else if (sparkIdentifier.namespace().length == 2) {
```

Comment:
what conditions when length == 2? could you add description ?

Reply:
for an Iceberg metadata table, such as `iceberg.default_db.test_iceberg_metadata_table.snapshots`, the `sparkIdentifier.namespace()` should be `['default_db', 'test_iceberg_metadata_table']`,  and there is a limitation in `RelationalCatalog`, so I decide to concact the `namespace` here to  avoid exception.

https://github.com/datastrato/gravitino/blob/ce29d83880a68c34f9aa25a8ced103c4ff9f118c/api/src/main/java/com/datastrato/gravitino/Namespace.java#L165-L170

Reply:
in this method, the Namespace will contain metalake, catalog and db.

Reply:
for `iceberg.default_db.test_iceberg_metadata_table.snapshots`, getDatabase will return `default_db.test_iceberg_metadata_table`? why not `default_db`?

Reply:
> for iceberg.default_db.test_iceberg_metadata_table.snapshots, getDatabase will return default_db.test_iceberg_metadata_table?

Yes.

> why not default_db?

for an iceberg metadata table, such as `iceberg.default_db.test_iceberg_metadata_table.snapshots`, the real table name is `snapshots`.

Reply:
> > for iceberg.default_db.test_iceberg_metadata_table.snapshots, getDatabase will return default_db.test_iceberg_metadata_table?
> 
> Yes.
> 
> > why not default_db?
> 
> for an iceberg metadata table, such as `iceberg.default_db.test_iceberg_metadata_table.snapshots`, the real table name is `snapshots`. Spark `Identifier` automatically uses the last part as a tableName, and everything else except the catalog and tableName as a namespace

cc: @FANNG1